src/antelope/stores/utils/currency-utils.ts

```diff
@@ -3,6 +3,23 @@ import { parseUnits } from 'ethers/lib/utils';
 import { formatUnits } from '@ethersproject/units';
 import Decimal from 'decimal.js';
 
+/**
+ * Given a number or string, returns a string representation of the number with up to 18 decimal places
+ * @param value - number or string to convert to string
+ * @returns {string} string representation of the number
+ */
+
+export function toStringNumber(value: number | string): string {
+    if (typeof value === 'string') {
+        return value;
+    } else if (typeof value === 'number') {
+        return value.toFixed(18);
```

Comment:
this will cause issues, as js numbers cannot handle 18 places of precision (max is 16). instead use this:
```ts
const num = new Decimal(value);
const result = num.toFixed(18);
```

Reply:
we already have the Decimal library on the project

Reply:
PERFECT !!!! :+1: